subprojects/testing-native/src/main/java/org/gradle/nativeplatform/test/xctest/plugins/XCTestConventionPlugin.java

```diff
@@ -84,13 +83,13 @@ public void apply(final ProjectInternal project) {
         final DirectoryVar buildDirectory = project.getLayout().getBuildDirectory();
         ConfigurationContainer configurations = project.getConfigurations();
         TaskContainer tasks = project.getTasks();
-        ProviderFactory providers = project.getProviders();
 
         // TODO - Reuse logic from Swift*Plugin
         // TODO - component name and extension name aren't the same
         // TODO - should use `src/xctext/swift` as the convention?
         // Add the component extension
-        SwiftComponent component = project.getExtensions().create(SwiftComponent.class, "xctest", DefaultSwiftComponent.class, "test", fileOperations, providers);
+        SwiftXCTestBundle component = objectFactory.newInstance(DefaultSwiftXCTestBundle.class, "test");
+        project.getExtensions().add(SwiftXCTestBundle.class, "test", component);
```

Comment:
I think the extension should be called `xctest` (as it was) rather than `test`. Ignore the fact that the component's name is `test` for now. It should be `xctest` but isn't yet.